### ISSUE TITLE
dynamically choosing the instrumentation ?

### ISSUE THREAD

**Issue by teytaud** [facebookresearch/nevergrad#798]
This one looks technically complicated.
We choose the instrumentation as part of the objective function.
Maybe the optimization algorithm might decide to change the instrumentation ?
For example, depending on the arity of a discrete domain, we might prefer Softmax or UnorderedChoice.

**Comment by jrapin** [comment]
I agree, still thinking about it and no solution yet. The thing is that you may also want to enforce the instrumentation as a user (typically for running expriments) so we would need to support both.
#513

**Comment by teytaud** [comment]
Agree... maybe we do not change anything in the structure of the codebase, and simply the optimizer can detect the instrumentation, creates another instrumentation, and converts data in order to adapt...

**Comment by jrapin** [comment]
> Agree... maybe we do not change anything in the structure of the codebase, and simply the optimizer can detect the instrumentation, creates another instrumentation, and converts data in order to adapt...

This would be hard to manage and can be confusing for a user :s but that's probably what will happen eventually
